android/src/main/java/com/oblador/keychain/KeychainModule.java

```diff
@@ -133,10 +131,16 @@
   //region Initialization
 
   /** Default constructor. */
-  public KeychainModule(@NonNull final ReactApplicationContext reactContext) {
+  public KeychainModule(@NonNull final ReactApplicationContext reactContext, @NonNull final IDeviceFilter deviceFilter) {
     super(reactContext);
     prefsStorage = new PrefsStorage(reactContext);
 
+  // Exclude the refactored code from the execution path to keep it aligned with the old version as much as possible.
+  // NOTE: This may result in reduction of the encryption capabilities.
+  // TODO Unwrap this workaround once the problem is resolved
+  if (deviceFilter.isDeviceAffected()) {
+    // TODO Add a fallback Cipher
+  } else {
```

Comment:
This is the actual workaround; the rest is just preparation and harness around it.

The fallback cipher is TBD.